src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -102,7 +103,16 @@ void ReadHeader()
         /// The number of columns in the row. -1 if there are no further rows.
         /// Note: This will currently be the same value for all rows, but this may change in the future.
         /// </returns>
-        public int StartRow()
+        public int StartRow() => StartRow(false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Starts reading a single row, must be invoked before reading any columns.
+        /// </summary>
+        /// <returns>
+        /// The number of columns in the row. -1 if there are no further rows.
+        /// Note: This will currently be the same value for all rows, but this may change in the future.
+        /// </returns>
+        public async Task<int> StartRow(bool async)
```

Comment:
We should not be publicly exposing methods that accept as async flag - that's an internal implementation detail. The pattern we use is to have public `X()` and `XAsync()`, both calling a private `X(bool async)` which contains the implementation.

Applies for the rest of the methods as well.